src/App.tsx

```diff
@@ -1,18 +1,37 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42)
   .map(n => `Item ${n}`);
 
+function getVisibleItems(
+  elements: string[],
+  currentPage: number,
+  perPage: number,
+): string[] {
+  const endIndex = perPage * currentPage;
+  const startIndex = endIndex - perPage;
+
+  return elements.slice(startIndex, endIndex);
+}
+
 export const App: React.FC = () => {
+  const [perPage, setPerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
```

Comment:
You can use one state for perPage and currentPage ( not critical, recommendation)